### PR TITLE
Add stop headsigns

### PR DESCRIPTION
**Pull request by philip-cline**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds support for stop_headsigns, which can be created at the PatternStop level. This PR requires backend PR #:  https://github.com/conveyal/gtfs-lib/pull/369

### CODE REVIEW

File: lib/editor/components/pattern/PatternStopCard.js [hunk 1]
```diff
@@ -439,6 +451,25 @@ class PatternStopContents extends Component<Props, State> {
...
+              <ControlLabel
+                // TODO: internationalize this
+                title={'Headsign that overrides trip headsign between stops.'}
+                className='small'>Stop headsign</ControlLabel>
+              <FormControl
+                type='text'
+                // TODO: internationalize this
+                placeholder={'Outbound'}
+                onChange={this._onHeadsignChange}
+                value={patternStop.stopHeadsign || ''}
```
Comment (binh-dam-ibigroup) [inline lib/editor/components/pattern/PatternStopCard.js]: Sort props

File: lib/editor/components/pattern/PatternStopCard.js [hunk 2]
```diff
@@ -439,6 +451,25 @@ class PatternStopContents extends Component<Props, State> {
...
+              controlId='stopHeadsign'
+              bsSize='small'>
+              <ControlLabel
+                // TODO: internationalize this
+                title={'Headsign that overrides trip headsign between stops.'}
+                className='small'>Stop headsign</ControlLabel>
+              <FormControl
+                type='text'
+                // TODO: internationalize this
+                placeholder={'Outbound'}
```
Comment (binh-dam-ibigroup) [inline lib/editor/components/pattern/PatternStopCard.js:467]: Nit: Is there are better way to convey that headsigns defined here are used in limited circumstances?

File: lib/editor/util/map.js [hunk 3]
```diff
@@ -769,7 +769,8 @@ export function stopToPatternStop (
     dropOffType: 0,
     pickupType: 0,
     shapeDistTraveled: null,
-    timepoint: null
+    timepoint: null,
+    stopHeadsign: ''
```
Comment (binh-dam-ibigroup) [inline lib/editor/util/map.js]: Sort attributes

File: lib/types/index.js [hunk 4]
```diff
@@ -595,11 +595,12 @@ export type PatternStop = {
   id: string | number,
   pickupType: number,
   shapeDistTraveled: ?number,
+  stopHeadsign: ?string,
   stopId: string,
   stopSequence: number,
   stop_lat?: number,
   stop_lon?: number,
-  timepoint: ?number
+  timepoint: ?number,
```
Comment (binh-dam-ibigroup) [inline lib/types/index.js]: Revert this change

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM after quick cleanup.

**Comment by binh-dam-ibigroup** [comment]
> LGTM after quick cleanup.

Although @philip-cline you probably need to fix the E2E tests for trip patterns.

File: lib/editor/components/pattern/PatternStopCard.js (hunk 2 above)
Comment (philip-cline) [reply lib/editor/components/pattern/PatternStopCard.js:467]: @binh-dam-ibigroup Thoughts on something like `Stop-specific headsign` as the placeholder text?

File: lib/editor/components/pattern/PatternStopCard.js (hunk 1 above)
Comment (philip-cline) [reply lib/editor/components/pattern/PatternStopCard.js]: Sorted in https://github.com/ibi-group/datatools-ui/pull/890/commits/ddf848719158b3500fd6520202ec781d5302dd5d

File: lib/editor/util/map.js (hunk 3 above)
Comment (philip-cline) [reply lib/editor/util/map.js]: Sorted in https://github.com/ibi-group/datatools-ui/pull/890/commits/ddf848719158b3500fd6520202ec781d5302dd5d

File: lib/types/index.js (hunk 4 above)
Comment (philip-cline) [reply lib/types/index.js]: Reverted in https://github.com/ibi-group/datatools-ui/pull/890/commits/ddf848719158b3500fd6520202ec781d5302dd5d

**Comment by philip-cline** [comment]
e2e tests are fixed by https://github.com/ibi-group/datatools-ui/pull/901

**Review by miles-grant-ibigroup — APPROVED** [review]
Works great thanks for the changes!
